### ISSUE TITLE
Validation Batch across GPUs during Training

### ISSUE THREAD

**Issue by bsvineethiitg** [xingyizhou/CenterNet#279]
I tried to make the validation faster during training by commenting the below line of code so that evaluation also happens by batches on Multiple GPUs.

https://github.com/xingyizhou/CenterNet/blob/a5a0483beb0f9e8705f6dc67d8817275369cfa7e/src/lib/trains/base_trainer.py#L50

But, this gives completely off loss values that are very high. Throughout the training, they remain high. Also, changing the batch sizes changes the validation loss strangely.

Moreover, reloading a trained model and running validation or training samples under .eval() model gives large losses.

Did you observe similar things? 

The problem seems to be related to evaluating on multiple GPUs using dataparallel when layers include batchnorm.

Any experience you had with this issue will help me. Thanks!

Edit: I had torch.backends.cudnn.enabled Enabled for BatchNorm. Looks like this is the issue?

**Comment by charliess123** [comment]
Hi, did you solve this problem. I have same issue when training with 'torch.backends.cudnn.enabled = True'
